### PR TITLE
[TIR] Add CUDA int4 tensor core intrinsics

### PR DESCRIPTION
**Pull request by vinx13**
This PR added int4 tensor intrinsic for CUDA tensor core.

cc @junrushao @tqchen @masahi

### CODE REVIEW

**Review by Hzfengsy — APPROVED** [review]
LGTM. But I want to remind you that the int4 Tensor Core support is removed from the 4th Tensor Core (Rtx 40 serious and Hopper)

File: python/tvm/tir/tensor_intrin/cuda.py [hunk 1]
```diff
@@ -817,6 +916,12 @@ def wmma_sync_impl(a: T.handle, b: T.handle, c: T.handle) -> None:
     *get_wmma_sync_intrin(16, 16, 16, "int8", "int32", True),
 )
 
+WMMA_SYNC_8x8x32_s4s4s32_TRANS_INTRIN = "wmma_sync_8x8x32_s4s4s32_trans"
```
Comment (yzh119) [inline python/tvm/tir/tensor_intrin/cuda.py:919]: "wmma_sync_8x8x32_s4s4s32" is missing.

**Review by yzh119 — COMMENTED** [review]
A slight issue, otherwise LGTM.

File: python/tvm/tir/tensor_intrin/cuda.py (hunk 1 above)
Comment (vinx13) [reply python/tvm/tir/tensor_intrin/cuda.py:919]: sub-byte tensor core only allows A in row major and B in col major

Comment (yzh119) [reply python/tvm/tir/tensor_intrin/cuda.py:919]: Oh that's interesting!

**Pull request merged**